packages/components/src/components/hds/form/key-value-inputs/index.ts

```diff
@@ -0,0 +1,179 @@
+  };
+  Blocks: {
+    header?: [
+      {
+        Legend?: WithBoundArgs<
+          typeof HdsFormLegendComponent,
+          'contextualClass' | 'isOptional' | 'isRequired' | 'id'
+        >;
+        HelperText?: WithBoundArgs<
+          typeof HdsFormHelperTextComponent,
+          'contextualClass' | 'controlId' | 'onInsert'
+        >;
+        Generic?: ComponentLike<HdsYieldSignature>;
+      },
+    ];
+    row: [
+      {
+        Field?: WithBoundArgs<
+          typeof HdsFormKeyValueInputsFieldComponent,
+          'rowIndex' | 'onInsert' | 'onRemove'
+        >;
+        DeleteRowButton?: WithBoundArgs<
+          typeof HdsFormKeyValueInputsDeleteRowButtonComponent,
+          'rowIndex' | 'rowData' | 'canDeleteRow'
+        >;
+        Generic?: WithBoundArgs<
+          typeof HdsFormKeyValueInputsGenericComponent,
+          'rowIndex' | 'onInsert' | 'onRemove'
+        >;
+        rowData?: unknown;
+        rowIndex?: number;
+      },
+    ];
+    footer?: [
+      {
+        Alert?: WithBoundArgs<typeof HdsAlertComponent, 'color' | 'type'>;
+        AddRowButton?: ComponentLike<HdsFormKeyValueInputsAddRowButtonSignature>;
+        Error?: WithBoundArgs<
+          typeof HdsFormErrorComponent,
+          'contextualClass' | 'controlId' | 'onInsert' | 'onRemove'
+        >;
+      },
+    ];
+  };
+  Element: HdsFormFieldsetSignature['Element'];
+}
+
+// @ts-expect-error: decorator function return type 'ClassOf<AriaDescribedBy>' is not assignable to 'typeof HdsFormField'
+@ariaDescribedBy
+export default class HdsFormKeyValueInputs extends Component<HdsFormKeyValueInputsSignature> {
+  private _id = guidFor(this);
+  private _element!: HTMLElement;
+  @tracked _gridTemplateColumns = '';
+
+  get canDeleteRow(): boolean {
+    return this.args.data.length > 1;
+  }
+
+  @action _setUpColumn(): void {
+    // eslint-disable-next-line ember/no-runloop
+    schedule('afterRender', (): void => {
+      this._updateColumns();
+    });
+  }
+
+  @action _removeColumn(): void {
+    // eslint-disable-next-line ember/no-runloop
+    schedule('afterRender', (): void => {
+      this._updateColumns();
+    });
+  }
+
+  @action
+  appendDescriptor(element: HTMLElement): void {
+    registerAriaDescriptionElement(this as AriaDescribedByComponent, element);
+  }
+
+  @action removeDescriptor(element: HTMLElement): void {
+    unregisterAriaDescriptionElement(this as AriaDescribedByComponent, element);
+  }
+
+  private _setUpKeyValueInputs = modifier((element: HTMLElement) => {
+    this._element = element;
+  });
+
+  // Update the column array based on how they are ordered in the DOM
+  private _updateColumns = () => {
+    const columns = this._element
+      .querySelector(KEY_VALUE_PAIR_FIRST_ROW_SELECTOR)
+      ?.querySelectorAll(
+        `${KEY_VALUE_PAIR_FIELD_SELECTOR}, ${KEY_VALUE_PAIR_GENERIC_SELECTOR}, ${KEY_VALUE_PAIR_DELETE_ROW_CONTAINER_SELECTOR}`
+      );
+
+    let newGridTemplateColumns = '';
```

Comment:
[thought] what do you think of this?
```suggestion
    let updatedGridTemplateColumns = '';
```